### PR TITLE
keypair generate: check if keypair doestn't exist for genrate one

### PR DESCRIPTION
**Pull request by keevBush**
We check if error is File not found before generate a keypair

### CODE REVIEW

File: cmd/stdiscosrv/main.go [hunk 1]
```diff
@@ -99,10 +99,12 @@ func main() {
 
 	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
-		log.Println("Failed to load keypair. Generating one, this might take a while...")
-		cert, err = tlsutil.NewCertificate(certFile, keyFile, "stdiscosrv", 20*365)
-		if err != nil {
-			log.Fatalln("Failed to generate X509 key pair:", err)
+		if os.IsNotExist(err) {
```
Comment (calmh) [inline cmd/stdiscosrv/main.go]: We still need to handle the case when there is an error but it's not "doesn't exist". I suggest 

```
if os.IsNotExist(err) {
  ... what you did
} else if err != nil {
  log.Fatalln("Failed to load keypair:", err
}
```

File: cmd/stdiscosrv/main.go [hunk 2]
```diff
@@ -99,10 +99,14 @@ func main() {
...
+			log.Println("Failed to load keypair. Generating one, this might take a while...")
+			cert, err = tlsutil.NewCertificate(certFile, keyFile, "stdiscosrv", 20*365)
+			if err != nil {
+				log.Fatalln("Failed to generate X509 key pair:", err)
+			}
+		} else if err != nil {
```
Comment (calmh) [inline cmd/stdiscosrv/main.go]: 👍 now we don't need the outer `if err != nil` either

Comment (keevBush) [reply cmd/stdiscosrv/main.go]: That error is for if we have another error with regenrate a keypair

Comment (calmh) [reply cmd/stdiscosrv/main.go]: Right. But `os.IsNotExist(nil)` is legal, and becomes false. So consider whether we need the extra `if err != nil` before it.

**Comment by calmh** [comment]
I think my meaning got lost. Please consider each branch and what will happen if the operation fails (`err != nil`). Currently an error to generate the certificate will go unreported and we'll crash when `cert` is nil. Something like this should be clean and simple:

```
cert, err := tls.LoadX509KeyPair(certFile, keyFile)
if os.IsNotExist(err) {
  // there is no certificate, we should create one, set same err variable
  cert, err = tlsutil.NewCertificate(certFile, keyFile, "stdiscosrv", 20*365)
}
if err != nil {
  // The error from LoadX509KeyPair was not "does not exist", or we got an error from NewCertificate
  log.Fatal(...)
}
```

**Comment by calmh** [comment]
I fixed this up to illustrate the minimal fix I was aiming for

**Pull request merged**